doc/cabal-commands.rst

```diff
@@ -142,6 +182,55 @@ arguments and flags that are common to some or all of them.
     Already generated `build-info.json` files will be removed since they would be stale otherwise.
 
 
+cabal list-bin
+--------------
+
+``cabal list-bin`` will either (a) display the path for a single executable or (b)
+complain that the target doesn't resolve to a single binary. In the latter case,
+it will name the binary products contained in the package. These products can 
+be used to narrow the search and get an actual path to a particular executable.
+
+Example showing a failure to resolve to a single executable.
+
+::
+
+    $ cabal list-bin cabal-install
+    cabal: The list-bin command is for finding a single binary at once. The
+    target 'cabal-install' refers to the package cabal-install-#.#.#.# which
+    includes the executable 'cabal', the test suite 'unit-tests', the test suite
+    'memory-usage-tests', the test suite 'long-tests' and the test suite
+    'integration-tests2'.
+
+For a scope that results in only one item we'll get a path.
+
+::
+
+    $ cabal list-bin cabal-install:exes
+    /.../dist-newstyle/build/.../cabal/cabal
+
+    $ cabal list-bin cabal-install:cabal
+    /.../dist-newstyle/build/.../cabal/cabal
+
+.. warning::
+    
+    ``list-bin`` won't accept any ``all`` scope such as ``all:exes`` or
```

Comment:
Actually, I think @fgaz recently said somewhere that `cabal list-bin all:exes` works if there is only one exe and I just tried it and it does work. Probably doesn't if there are more (or zero?) exes.

Reply:
Thanks for the catch. I was trying out the behaviour of list-bin with cabal itself, a big `cabal.project`. I should review the tests for list-bin too while I'm on it.

Reply:
I got `cabal list-bin all` and `cabal list-bin all:exes` to work on some other projects too so I removed the warning. @Mikolaj can you please review this for merge now?